src/runtime/virtcontainers/sandbox.go

```diff
@@ -492,6 +498,10 @@ func createSandbox(ctx context.Context, sandboxConfig SandboxConfig, factory Fac
 		return s, nil
 	}
 
+	if err := s.fsShare.Prepare(ctx); err != nil {
+		return nil, err
+	}
+
 	// Below code path is called only during create, because of earlier Check.
```

Comment:
should this comment move above the prepare (and could be clarified?). 

Either way, afaiu, we  only want to  setup the shared filesystem if we are creating a new sandbox, instead of loading the sandbox state, correct?

Reply:
I clarified and moved the comment, thanks.

Also, I changed the `Prepare` implementation to make it idempotent, and it is now safe to call it several times in a row.